### PR TITLE
ts-web: parsing a signature request

### PR DESCRIPTION
**Pull request by pro-wh**
this will be useful for the extension side when it gets a ContextSigner.sign request

### CODE REVIEW

**Comment by pro-wh** [comment]
Here's a draft of how to parse a signature request into a transaction and then parse a transaction into a specific method.

You'd first call `oasis.signature.visitMessage(handlers, context, message)`. That looks at the `context` string to figure out what kind of message it is. It parses the message CBOR and calls one of the handlers. Use `SignatureMessageHandler*` types from the other modules to guide what context keys and handler types are available. Currently we're starting out with consensus transactions (`oasis.consensus.SignatureMessageHandlersWithChainContext`) and runtime transactions (`oasisRT.transaction.SignatureMessageHandlersWithChainContext`).

Then inside that, there are `oasis.consensus.visitTransaction(handlers, tx)` for consensus transactions and `oasisRT.transaction.visitCall(handlers, call)` for runtime transactions. Use the `ConsensusTransactionHandlers` and `TransactionCallHandlers` respectively from other modules to guide what method keys and handler types are available.

**Comment by pro-wh** [comment]
setting target branch to main for CI

**Review by kostko — COMMENTED** [review]
There seem to be some issues with the `e2e-ts-web-ext-utils` test?

**Comment by pro-wh** [comment]
oh I hadn't added a step to prepare the rt package. added that now

**Review by kostko — APPROVED** [review]

File: client-sdk/ts-web/ext-utils/sample-ext/src/index.js [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+                                            confMessage += `
+Recognized method: staking allow
+Beneficiary: ${oasis.staking.addressToBech32(body.beneficiary)}
+Amount change: ${body.negative ? '-' : '+'}${oasis.quantity.toBigInt(
+                                                body.amount_change,
+                                            )} base units`;
+                                        },
+                                        [oasis.staking.METHOD_WITHDRAW]: (body) => {
+                                            confMessage += `
+Recognized method: staking allow
```
Comment (tjanez) [inline client-sdk/ts-web/ext-utils/sample-ext/src/index.js:186]:
```suggestion
Recognized method: staking withdraw
```

**Comment by pro-wh** [comment]
thanks for looking through. gonna merge

**Pull request merged**